users/views.py

```diff
@@ -1,3 +1,31 @@
-from django.shortcuts import render
+import json, jwt, requests, random
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from users.models import User
+from my_settings  import SECRET_KEY
+
+class KakaoSignInView(View): 
+    def get(self, request): 
+        try:
+            access_token = request.headers.get('Authorization')
+            print(access_token)
+            if not access_token: 
+                return JsonResponse({'message':'no_token'}, status=401)
+
+            url      = 'https://kapi.kakao.com/v2/user/me'
+            response = requests.get(url, headers={'Authorization': f'Bearer {access_token}'})
+            user_response = response.json()
+        
+            user,created = User.objects.get_or_create(kakao_id=user_response['id'])
+            if created: 
+                user.email    = user_response['kakao_account']['email']
+                user.nickname = user_response['properties']['nickname']
+                user.point    = random.randrange(0,10000)
+                user.save()
+
+            token = jwt.encode({'id':user.id}, SECRET_KEY, algorithm= 'HS256')
+        except KeyError:
+            return JsonResponse({'message':'invalid_token'}, status=400)
+        return JsonResponse({'TOKEN':token}, status=200)
```

Comment:
```suggestion
            return JsonResponse({'TOKEN':token}, status=200)
         except KeyError:
            return JsonResponse({'message':'invalid_token'}, status=400)
       
```
성공 케이스를 except 위로 올리고 except들은 모두 아래로 내리는 것이 흐름상 더 자연스럽고 가독서에 더 좋습니다.

Reply:
수정했습니다!!